### ISSUE TITLE
closing tag and opening tag combination going to the for loop.

### ISSUE THREAD

**Issue by sk29110** [Tarik02/html-substring-js#11]
I have test "><" and falling into fatal error.

![image](https://user-images.githubusercontent.com/626182/61556205-2350a180-aa16-11e9-9d4e-1279f178083c.png)

my original test case qa put was ">>>``\_(*L*)_/``<<>>>``\_(*L*)_/``<< "

**Comment by tarik02** [comment]
The problem appears whenever we have an unclosed tag start, so. New testcases draft:
```typescript
  describe('unclosed tag name', () => {
    it('should work with \'<\' as input', () => {
      const result = html_substring('<', 50);

      expect(result).to.eq('<');
    });

    it('should work with \'<div\' as input', () => {
      const result = html_substring('<div', 50);

      expect(result).to.eq('<div');
    });
    
    it('should work with \'<div abc\' as input', () => {
      const result = html_substring('<div abc', 50);

      expect(result).to.eq('<div abc');
    });

    it('should work with \'test <div abc\' as input', () => {
      const result = html_substring('test <div abc', 50);

      expect(result).to.eq('<test div abc');
    });

    it('should work with \'hello <div <h1\' as input', () => {
      const result = html_substring('test hello <div <h1', 50);

      expect(result).to.eq('hello <div <h1');
    });

    it('should work with \'<div<h1>\' as input', () => {
      const result = html_substring('<div<h1>', 50);

      expect(result).to.eq('<div<h1>');
    });
  });
```

**Comment by tarik02** [comment]
Fixed by [8135e2f](https://github.com/Tarik02/html-substring-js/commit/8135e2f55b236a86d6dd7fc233da2a8e68892423) (starting with version 1.8.0)
